implementations/rust/ockam/ockam_api/src/cloud/share/create.rs

```diff
@@ -0,0 +1,163 @@
+use minicbor::{Decode, Encode};
+use serde::Serialize;
+
+use super::*;
+
+#[derive(Clone, Debug, Decode, Encode, Serialize)]
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct CreateInvite {
+    #[n(1)] pub expires_at: Option<String>,
+    #[n(2)] pub grant_role: RoleInShare,
+    #[n(3)] pub recipient_email: Option<String>,
+    #[n(4)] pub remaining_uses: Option<usize>,
+    #[n(5)] pub scope: ShareScope,
+    #[n(6)] pub target_id: String,
+}
+
+#[derive(Clone, Debug, Decode, Encode, Serialize)]
+#[cbor(map)]
+#[rustfmt::skip]
+pub struct CreateServiceInvite {
+    #[n(1)] pub expires_at: Option<String>,
+    #[n(2)] pub project_id: String,
+    #[n(3)] pub recipient_email: String,
+
+    // TODO: Is there an identity type I should use that may be CBOR-encoded?
+    // TODO: Should route be a MultiAddr?
+    #[n(4)] pub project_identity: String,
+    #[n(5)] pub project_route: String,
+    #[n(6)] pub project_authority_identity: String,
+    #[n(7)] pub project_authority_route: String,
+    #[n(8)] pub shared_node_identity: String,
+    #[n(9)] pub shared_node_route: String,
+}
+
+mod node {
+    use ockam_core::api::{Request, Response};
+    use ockam_core::{self, Result};
+    use ockam_multiaddr::MultiAddr;
+    use ockam_node::Context;
+    use tracing::trace;
+
+    use crate::cloud::CloudRequestWrapper;
+    use crate::nodes::{NodeManager, NodeManagerWorker};
+
+    use super::*;
+
+    const API_SERVICE: &str = "users";
+
+    impl NodeManager {
+        // TODO: Should identity_name not be an Option for this use-case?
+        //       This is an authenticated-only request
+        pub async fn create_invite(
+            &self,
+            ctx: &Context,
+            req: CreateInvite,
+            route: &MultiAddr,
+            identity_name: Option<String>,
```

Comment:
See my comment above. `identity_name` is a way to select the identity used to create a secure channel. It has no meaning outside of a local installation